build.gradle.kts

```diff
@@ -55,22 +55,41 @@ plugins {
     id("org.cadixdev.licenser") version("0.6.1")
 }
 
-configure<JavaPluginExtension> {
+java {
     sourceCompatibility = JavaVersion.VERSION_16
     targetCompatibility = JavaVersion.VERSION_16
+    withSourcesJar()
 }
 
 group = modGroup
 version = modVersion + getVersionDecoration()
 
-base {
-    archivesName.set(modName)
+base.archivesName.set(modName)
+
+val gametestSourceSet = sourceSets.create("gametest") {
+    compileClasspath += sourceSets.main.get().compileClasspath
+    compileClasspath += sourceSets.main.get().output
+    runtimeClasspath += sourceSets.main.get().runtimeClasspath
+    runtimeClasspath += sourceSets.main.get().output
+    java.srcDir("src/gametest/java")
+    resources.srcDir("src/gametest/resources")
 }
 
 loom {
     accessWidenerPath.set(project.file("src/main/resources/galacticraft.accesswidener"))
     mixin {
-        add("main", "galacticraft.refmap.json")
+        add(sourceSets.main.get(), "galacticraft.refmap.json")
+    }
+
+    runs {
+        register("gametest") {
```

Comment:
can we get a client gametest run too?

Reply:
for local testing in dev envs

Reply:
Added in 850f716c0818ed4267c7c1274ae7377874606fc8